### ISSUE TITLE
Cookie House flashes "Error" instead of the cookie name briefly when a purchase is made

### ISSUE THREAD

**Issue by NevilleS** [ethyca/fides#2138]
### Bug Description

See this Loom video to see this live: https://www.loom.com/share/2b5c6b85404e46dabb1a7ccf764460a5

When you submit a purchase and the modal starts to fade out, the title of the form temporarily switches to "Error". I didn't catch this when I was first reviewing this code!

Here's the offending screengrab:
<img width="572" alt="image" src="https://user-images.githubusercontent.com/1834295/210698885-3a82e3c5-4af4-4113-b341-a2c0d44dc778.png">

#### Steps to Reproduce

1. Run `fides deploy up` on any version of `fides` 2.x+
2. Open the Cookie House (http://localhost:3000)
3. Purchase any cookie 

To work on this really quickly though without needing a full `fides` release from pypi, you can build & run the sample Cookie House project from source... it's just not particularly easy to do! Here's how to do that:
1. Run `nox -s build(sample)` to rebuild all the sample project images: `fides-sample-app:sample`, `fides-privacy-center:sample`, etc. Fair warning: this can take a long time the first time you do it!
2. Run `cd src/fides/data/sample_project` to get into the sample project working directory
3. Run `FIDES_UPLOADS_DIR=. docker compose up`
4. Open http://localhost:3000

### Expected behavior

No "Error" text should be displayed. Instead, the form title can just be "Submit Your Order" instead 🍪 

### Environment

* Version: 2.4.0
* OS: macOS-12.6.2-x86_64-i386-64bit
* Python Version: 3.9.6
* Docker Version: 20.10.17

### Additional context

The good news here is that this should be a one-line change, once you just get familiar with this sample project setup (that's the hard part). See this code here: https://github.com/ethyca/fides/blob/836222c541648fabcf9bdcdb5e69f8666b56b46a/src/fides/data/sample_project/cookie_house/src/components/PurchaseModal/index.tsx#L58

Basically, the `product` prop can't be trusted to stick around when the modal is fading out. We should either:
1. Change the title of the modal to not require the `product` prop (e.g. just make the title "Submit Your Order")
2. Or, if you're feeling spicy, make the logic in `Home.tsx` that manages the `productInPurchase` state to be smarter about not wiping it out before the purchase is really finished! 😄

**Comment by LKCSmith** [comment]
@NevilleS - is there a trick to get UI updates when making changes to this part of the codebase? When running the project, none of the changes I make to the code are being reflected in the UI - it stays static.

**Comment by ssangervasi** [comment]
@LKCSmith I haven't worked on the sample project before, but my guess is that running with Neville's instructions requires a rebuild of the docker image with any changes. The [Dockerfile](https://github.com/ethyca/fides/blob/main/src/fides/data/sample_project/cookie_house/Dockerfile#L14) copies the source into the into the image and then builds it in there, so your edits aren't in there unless the image is rebuilt.

**Comment by NevilleS** [comment]
Yeah - it's not built for rapid development, sorry! The only way to try it out locally is to tear it down and re-run `nox -s build(sample)` to get those changes.

To get it to be more interactive, we'd need to edit the `docker-compose.yml` file that spins it up to run the app from source. That'd be possible but I think in this case you should be able to get this change implemented with only a couple builds...!

**Comment by LKCSmith** [comment]
Gotcha, thanks both of you. I could've sworn I tried rebuilding after making some changes and my changes were still not captured, but that was a little bit ago. I'll try again!
